### PR TITLE
Link TODOs to Issues

### PR DESCRIPTION
**Pull request by xsebek**
- replace `XXX` with `TODO: #<issue number>`

TODO has simply better editor support. The GitHub plugin will suggest creating an issue for TODO that does not have the issue number after it and some editors may even ask for confirmation when committing code with TODO.

### CODE REVIEW

**Review by byorgey — APPROVED** [review]
`XXX` is what I historically have put as a reminder to myself, but very happy to change to something more standard / with better tool support.  Linking `TODO`s explicitly to issues sounds like a great practice.

File: src/Swarm/TUI/View.hs [hunk 1]
```diff
@@ -603,7 +603,7 @@ drawWorld g =
           ixs = range (viewingRegion g (fromIntegral w, fromIntegral h))
       render . vBox . map hBox . chunksOf w . map drawLoc $ ixs
  where
-  -- XXX update how this works!  Gather all displays, all
+  -- TODO: update how this works!  Gather all displays, all
```
[the remark this answers is not in the document]
Comment (byorgey) [reply src/Swarm/TUI/View.hs:606]: It's really not very important, and at this point, I'm almost inclined to say "if it ain't broke, don't fix it".

**Comment by byorgey** [comment]
@xsebek , not all editors may warn about `TODO` etc. (I don't think mine does).  Is this something we can/should add to our CI? *i.e.* the CI will fail if there is any `XXX`, or any `TODO` without an issue number?

**Comment by byorgey** [comment]
Free form comment is great, I had in mind something like requiring `TODO: #nnn` but you can have whatever you want after that.  I don't think it would diacourage me from writing a comment.  Just an idea though.

**Comment by xsebek** [comment]
But we can also create our own action, for one line it just needs a regex and there are blog posts on how to do it.

**Comment by xsebek** [comment]
Well, the Haskell PR to the action was merged, we might as well give it a try 🙂

**Comment by xsebek** [comment]
Well, to use the action with the `TODO: #123` style, it might need some extra work, I filed an issue there:
 - alstr/todo-to-issue-action/issues/123

I will see if I can implement it there - instead of creating a new issue, it would add a comment to the linked one.

**Comment by byorgey** [comment]
@xsebek What's the status of this? I can think of three "levels" to which we could do this:

- Level 0: clean up any existing XXX and TODO in the code and link them to issues.
- Level 1: put something simple in our CI so it rejects any code containing XXX or containing TODO without an issue number.
- Level 2: use something like `todo-to-issue-action` to automate things even more.

It seems like we have stalled out trying to achieve Level 2. I think we should do Level 0 right away and then make separate issues for Level 1 and/or Level 2.

**Comment by xsebek** [comment]
@byorgey agreed, I will rebase and clean this up to solve "Level 0". 

Level 1 could be our own GitHub action that just rejects TODOs that do not match our regex - i.e. there has to be a linked open issue. Should be easy to do, possibly even in this PR.

I think level 2 is best solved as a separate Issue - ideally by improving an existing GitHub action and then adding it to Swarm.
We could write the logic ourselves - "get the additions from the last commit, if you find TODO with Issue number comment to that Issue" - but I think it would end up being more work.

**Review by byorgey — APPROVED** [review]
Looks good to me!

**Comment by xsebek** [comment]
Rebased and added a simple regex script that will check for TODOs without a linked Issue.

This should help us to get to level 1. :smile:

**Pull request merged**